### PR TITLE
Add YieldExpression

### PR DESCRIPTION
**Pull request by maximderbin**

### CODE REVIEW

File: src/nodes/function_expression.js [hunk 1]
```diff
@@ -34,6 +34,10 @@ export function format(node, context, recur, {noFunctionKeyword} = {}) {
         context.write('function');
     }
 
+    if (node.generator) {
+        context.write('* ');
```
Comment (dmitriiabramov) [inline src/nodes/function_expression.js]: why space after `*`?

Comment (maximderbin) [inline src/nodes/function_expression.js]: To match this one https://github.com/dmitriiabramov/esfmt/commit/f362b8546aa879a8b27d40774d07a8520746905e#diff-926a5400e756f7ac901a5cc4b77695f0R14

Comment (dmitriiabramov) [inline src/nodes/function_expression.js]: interesting... i would have these styles for generators

```js
funciton* a() {}

function *a() {}

let a = function*() {}
```

but i'm not sure which one would be more correct :)

Comment (maximderbin) [inline src/nodes/function_expression.js]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Iterators_and_Generators

```javascript
function* idMaker(){
  var index = 0;
  while(true)
    yield index++;
}

myIterable[Symbol.iterator] = function* () {
    yield 1;
    yield 2;
    yield 3;
};
```

Comment (maximderbin) [inline src/nodes/function_expression.js]: I took it from there

**Pull request merged**